app/services/metadata_cloud_service.rb

```diff
@@ -4,36 +4,78 @@
 class MetadataCloudService
   ##
   # This is the method that is called from the yale:refresh_fixture_data rake task
-  def self.refresh_fixture_data(oid_path)
+  def self.refresh_fixture_data(oid_path, metadata_source)
     mcs = MetadataCloudService.new
     mcs.list_of_oids(oid_path).each do |oid|
-      metadata_cloud_url = mcs.build_metadata_cloud_url(oid)
+      next unless mcs.build_metadata_cloud_url(oid, metadata_source)
+      metadata_cloud_url = mcs.build_metadata_cloud_url(oid, metadata_source)
       full_response = mcs.mc_get(metadata_cloud_url)
-      mcs.save_mc_json_to_file(full_response, oid)
+      mcs.save_mc_json_to_file(full_response, oid, metadata_source)
     end
   end
 
   ##
   # Takes a Metadata Cloud formatted url and returns the full HTTP response with headers
-  def mc_get(oid_url)
+  def mc_get(mc_url)
     metadata_cloud_username = ENV["MC_USER"]
     metadata_cloud_password = ENV["MC_PW"]
-    HTTP.basic_auth(user: metadata_cloud_username, pass: metadata_cloud_password).get(oid_url)
+    HTTP.basic_auth(user: metadata_cloud_username, pass: metadata_cloud_password).get(mc_url)
+  end
+
+  ##
+  # Takes an oid (Ladybird identifier) and a metadata source (allowed values are ladybird, ils, and aspace), and returns
+  # the appropriate URL to pull the metadata from the Yale Metadata Cloud
+  def build_metadata_cloud_url(oid, metadata_source)
```

Comment:
I'm going to merge this as-is but this method looks ripe for refactoring into three separate methods.